### ISSUE TITLE
💅 False positive of noAccudmulatingSpread on function parameter spread

### ISSUE THREAD

**Issue by unional** [biomejs/biome#2183]
### Environment information

```bash
CLI:
  Version:                      1.6.2
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           windows

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         unset
  JS_RUNTIME_VERSION:           "v18.19.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "pnpm/8.15.4"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    true
  VCS disabled:                 true

Linter:
  Recommended:                  true
  All:                          false
  Rules:                        complexity/noBannedTypes = "off"
                                complexity/noForEach = "off"
                                complexity/useArrowFunction = "warn"
                                complexity/useLiteralKeys = "off"
                                correctness/noUnusedVariables = "error"
                                correctness/useExhaustiveDependencies = "off"
                                style/noCommaOperator = "off"
                                style/noNonNullAssertion = "off"
                                style/noParameterAssign = "off"
                                style/noUnusedTemplateLiteral = "warn"
                                suspicious/noAssignInExpressions = "off"
                                suspicious/noConfusingVoidType = "off"
                                suspicious/noExplicitAny = "off"

Workspace:
  Open Documents:               0
```

### Rule name

noAccudmulatingSpread 

### Playground link

https://biomejs.dev/playground/?code=KAAuAC4ALgBhAHIAZwBzADoAIABzAHQAcgBpAG4AZwBbAF0AKQAgAD0APgAgAGYAbgBzAC4AcgBlAGQAdQBjAGUAKAAoAGEAcgBnAHMALAAgAGYAbgApACAAPQA%2BACAAWwBmAG4AKAAuAC4ALgBhAHIAZwBzACkAXQAsACAAYQByAGcAcwApAFsAMABdAA%3D%3D

### Expected result

It should not fail

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by Sec-ant** [comment]
I don't know very much about this specific use case, but it looks to me that this rule is doing [what it claims to do](https://biomejs.dev/linter/rules/no-accumulating-spread/):

> Disallow the use of spread (`...`) syntax on accumulators.

Maybe you should

- Suppress this rule in this specific use case:

  ```ts
  // biome-ignore lint/performance/noAccumulatingSpread: <explanation>
  ```

- Or change your code to something like this:

  ```ts
  (...args: string[]) => fns.reduce((args, fn) => [fn.apply(undefined, args)], args)[0]
  ```

**Comment by Conaclos** [comment]
I agree with @Sec-ant. It looks like the rule is reporting a problematic code.

Could you give us some real examples where this pattern is useful?

**Comment by unional** [comment]
Yes, you are correct. Originally I thought this could be a special case. But since `fn.apply()` does not affect already bound function, using `fn.apply(null/undefined, args)` is better.

**Comment by Sec-ant** [comment]
> since `fn.apply()` does not affect already bound function

I wasn't sure about this, thanks for share it!

**Comment by unional** [comment]
> I wasn't sure about this, thanks for share it!

Me neither. That's why I was writing it as `fn(...args)`. But your comment help me to actually test it out. 🍻
